libcontainer/cgroups/systemd/v2.go

```diff
@@ -34,6 +36,39 @@ func NewUnifiedManager(config *configs.Cgroup, path string, rootless bool) cgrou
 	}
 }
 
+// unifiedResToSystemdProps tries to convert from Cgroup.Resources.Unified
+// key/value map to systemd unit properties. This is on a best-effort basis,
+// i.e. all the unknown parameters are ignored.
+func unifiedResToSystemdProps(res map[string]string) (props []systemdDbus.Property, _ error) {
+	for k, v := range res {
+		if strings.Contains(k, "/") {
+			return nil, fmt.Errorf("unified resource %q must be a file name (no slashes)", k)
+		}
+		sk := strings.SplitN(k, ".", 2)
+		if len(sk) != 2 {
+			return nil, fmt.Errorf("unified resource %q must be in the form CONTROLLER.PARAMETER", k)
+		}
+		switch k {
+		case "pids.max":
+			num := uint64(math.MaxUint64)
+			if v != "max" {
+				var err error
+				num, err = strconv.ParseUint(v, 10, 64)
+				if err != nil {
+					return nil, fmt.Errorf("unified resource %q value conversion error: %w", k, err)
+				}
+			}
+			props = append(props,
+				newProp("TasksAccounting", true),
+				newProp("TasksMax", uint64(num)))
+		default:
+			logrus.Debugf("skipping unknown unified resource %q=%q", k, v)
```

Comment:
@giuseppe 

Should this be an error?

Reply:
I believe yes.  Otherwise the runtime will ignore the user configuration.  Alternatively, if it is not handled by systemd, should it be written directly to the cgroupfs?

Reply:
I don't think so. This is a "cgroupfs v2 file -> systemd unit property" translation, done on a best effort basis. We can't possibly know about and translate all the properties.

The fs2 cgroup driver will be called later and it will return an error in case a parameter file does not exist.

Reply:
Still I left a debug message so we know the property is ignored (by the translation layer). Maybe the message should be amended to make it clear that this is just for the systemd property, not for the parameter per se.

Reply:
To reiterate, runc systemd driver always uses fs/fs2 driver as a backend -- first it creates everything using systemd means, when it calls fs/fs2's Set. This is ugly, but that's the way it is. One reason, I guess, is that systemd does not support all cgroupfs controllers/parameters.

Reply:
SGTM

Reply:
I have improved the debug message, and added a comment explaining the resource won't be ignored.